doc/adr/0003-dag-l1-responsibility-and-l0-interconnection.md

```diff
@@ -0,0 +1,30 @@
+# 3. DAG L1 responsibility and L0 interconnection
+
+Date: 2022-01-05
+
+## Status
+
+Proposed
+
+## Context
+
+DAG L1 responsibility and communication between DAG L1 and L0 needs to be
+specified.
+
+## Decision
+
+DAG Layer 1 returns DAG blocks as an output and it doesn't create snapshots.
+DAG Layer 1 sends DAG blocks to the Layer 0 so then the Layer 0 can create a Global
+snapshot.
+
+Cross-chain swaps will take place on the Layer 0, so then Layer 0 can create
+additional DAG transactions and blocks.
```

Comment:
The actual data types for the swap, called Emission currently, should be enough to resister transfer across addresses/liquidity pools. However, this will happen at the L0 level

Reply:
What we mean is that DAG transactions and blocks will not only be created on the L1 but also on L0.
And due to no direct communication (push) from L0 to L1, both these things imply that when L1 will download a new global snapshot, there can be blocks/transactions new (unknown) for L1 so the L1 needs to reapply them (or simply apply the whole global snapshot and skip already accepted data)

Reply:
Sure, yes L0 needs to create a block for the base/timed trigger case. Also Transactions for Rosetta standard rewards etc.

Reply:
That's fine. L0 nodes can make similar consensus rounds as DAG L1 nodes.
cc: @TheMMaciek

Reply:
Well, the L0 cells don't need to run a full L1 consensus to make an "empty" block for a snapshot, in the case there are no new blocks. We can consider a snapshot/block (which has its two tips being the edge of the previous snapshot) as a special case, just for snapshot proposals.

However, nodes runing L0 can run L1 cells as well, this is where the Arrow comes in

Reply:
I didn't mean an empty block but a normal consensus round, to process additional transactions on L0 (eg. from cross-chain swaps)

Reply:
We shouldn't be running L1 consensus via L0. If we need to perform a swap, an emission from L1 should be enough to initiate the transfer